piksi_tools/console/console.py

```diff
@@ -787,128 +786,131 @@ def __init__(self, usage, error_str):
         default_button='OK',
         close_result=False,
+        resizable=True,
         icon=icon,
-        width=460,
-        title='Swift Console v{0} - Select Interface'.format(CONSOLE_VERSION)
+        title="Swift Console Encountered An Error",
     )
 
-    def refresh_ports(self):
-        """
-        This method refreshes the port list
-        """
-        try:
-            self.ports = [p for p, _, _ in s.get_ports()]
-        except TypeError:
-            pass
+    def __init__(self, error_message=None, e=None):
+        self.error_str = (error_message
+                          if error_message is not None
+                          else 'Unhandled Error')
+        if e is not None:
+            self.error_str += '\n\n{}'.format(e)
 
-    def _refresh_ports_button_fired(self):
-        self.refresh_ports()
 
-    def __init__(self, baudrate=None):
-        self.refresh_ports()
-        # As default value, use the first city in the list:
+class ConnectionData(object):
+    '''Data class for connection information.'''
+    def __init__(self, driver=None, description="", mode=None,
+                 connection_info={}):
+        self.driver = driver
+        self.description = description
+        self.connection_info = connection_info
+        if mode is not None:
+            self.set_connection_mode(mode)
+
+    def set_connection_mode(self, mode):
+        self.connection_info['mode'] = mode
+
+
+def connection_from_args(args):
+    cnx_data = None
+    connect_error = None
+    description = ''
+    mode = ''
+    port = getattr(args, 'port', None)
+    baud = getattr(args, 'baud', None)
+
+    if port is not None:
+        if args.tcp is True:
+            # Use the TPC driver and interpret port arg as host:port
+            description = port
+            mode = 'TCP/IP'
+        elif args.file is True:
+            # Use file and interpret port arg as the file
+            print("Using file '%s'" % port)
+            description = os.path.split(port)[-1]
+            mode = 'file'
+        else:
+            if args.rtscts is True:
+                print("using flow control")
+            # Use the port passed and assume serial connection
+            print("Using serial device '%s'" % port)
+            description = os.path.split(port)[-1]
+            description += (" @" + str(baud)) if baud else ""
+            mode = 'serial'
+
         try:
-            self.port = self.ports[0]
-        except IndexError:
-            pass
-        if baudrate not in BAUD_LIST:
-            self.choose_baud = False
-        self.baudrate = baudrate
+            driver = s.get_base_args_driver(args)
+        except Exception as e:  # noqa
+            connect_error = e
+        else:
+            cnx_data = ConnectionData(
+                driver=driver,
+                description=description,
+                mode=mode
+            )
+    return (cnx_data, connect_error)
+
+
+def port_chooser_retry_logic(args):
```

Comment:
Should theese be in the `port_chooser.py`?

Reply:
I left it out as it uses `connection_from_args`, basically it the how-console-uses-port-chooser method